dhcpcd-hooks/65-kano-dhcp-hook

```diff
@@ -38,24 +36,60 @@ REBOOT | BOUND)
     # Try uploading the tracking data to our servers
     # Should be quiet on failure
     kano-sync --upload-tracking-data --silent &
+}
 
-    # Sync objects from the content API in the background
-    # TODO: This should be removed once we have the daemon done
+# Sync objects from the content API in the background
+# TODO: This should be removed once we have the daemon done
+kano_daemon () {
     kano-content sync &
-
     kano-tracker-ctl +1 'internet-connection-established' &
-    ;;
+}
 
-NOCARRIER | STOPPED)
-    # Are we still networked due to multihomed connection?
+# Check for a multihomed connection and keep it if there
+# otherwise disconnect with a logger notification
+is_multihomed () {
     is_internet
     if [ $? -eq 0 ]; then
-    	logger "kano-dhcp-hook: NOCARRIER/STOPPED but still connected, aborting action"
+    	logger "kano-dhcp-hook: $reason but still connected, aborting action."
     	exit 0
     else
-        logger "kano-dhcp-hook: NOCARRIER/STOPPED and connectivity lost"
+        logger "kano-dhcp-hook: $reason and connectivity lost."
         rm $monitor_file
     fi
+}
+
+# In case of lease RENEWal or REBOUND to a new DHCP server
+# check if connection is sucesfully active and modifly
+# the monitor_file accordingly
+is_reconnected () {
+    if [$? -eq 0]; then
+      logger "kano-dhcp-hook: $reason but still connected...Aborting."
+    exit 0
```

Comment:
There's a catch with these hook scripts: you cannot use `exit` otherwise all subsequent hooks will not execute.

Reply:
Yup, I am using `exit` in all cases that there is already connection present, that means hooks don't need to be triggered and the `wpa_supplicant` handles the connection as it is. Actually, there is no change there from the previous script, I only created the functions for a clearer structure.

Reply:
Oh, I see. I'm not really happy with breaking the event listener chain as I can't see what gotchas there may be. If that is who it's going to be handled, please clearly document this behaviour in the header of the file.